torch/csrc/distributed/c10d/init.cpp

```diff
@@ -1461,7 +1462,12 @@ Example::
                 ``fut.then()`` will return another ``CUDAFuture`` that holds the return value of the
                 callback and a ``CUDAEvent`` that recorded the callback stream.
 
-                Note that ``fut.done()`` returns only whether the operation has been enqueued on the GPU.
+                For CPU work, ``fut.done()`` returns true when work has been complted and value()
```

Comment:
Better to itemize these description in the same paragraph:
1) For CPU work, ...
2) For GPU work, ...
3) For mixed CPU-GPU work, ...

When this docstring is included by an rst file and displayed as a webpage, this can give a more clear view.